bagua/torch_api/data_parallel/bagua_distributed.py

```diff
@@ -0,0 +1,478 @@
+# pytype: disable=attribute-error
+import torch
+import time
+import io
+import pickle
+import collections
+import logging
+from typing import List, Tuple, Optional
+from torch.nn.modules import Module
+
+import bagua
+from bagua.torch_api import env
+from bagua.torch_api.communication import (
+    get_backend,
+    get_hyperparameters_service_client,
+    broadcast,
+    BaguaProcessGroup,
+)
+from bagua.torch_api.model_parallel.moe import is_moe_param
+from bagua.bagua_define import (
+    TensorDeclaration,
+    BaguaHyperparameter,
+)
+from bagua.torch_api.utils import to_bagua_datatype, StatisticalAverage
+
+
+class BaguaPatches:
```

Comment:
How about making this `BaguaDistributedDataParallel.bagua_states` attribute instead of a dangling class